src/sqlfe/general/Submission.java

```diff
@@ -405,6 +367,74 @@ else if (Utilities.isUserCommentSingleLine(line) ||
 
 	}	// end - method readSubmission
 	
+	// Get the details of the file and store them in submissionName and studentName
+	private String getFileMetadata(BufferedReader br) {
+		// TODO - remove -- -- from these three lines before writing out
+		// TODO - make how many lines are in an assignment customizable? e.g. add student id
+		String line = null;
+		try {
+			line = br.readLine(); 											// get first line
+			line = Utilities.skipBlankLines(br, line);						// skip blanks if any
+			final int BASE_PROMPT_LENGTH = 6;								// length of instructor comment marker >-- -- <
+			submissionName = line.substring(BASE_PROMPT_LENGTH);			// first line = assignment name, strip off leading >-- -- <
+			//System.out.println("submission name: " + submissionName);
+			line = br.readLine();											// second line = (student) name
+			final int NAME_PROMPT_LENGTH = 11;								// length of >-- -- Name:<
+			if (line.length() > NAME_PROMPT_LENGTH) {
+				studentName = line.substring(NAME_PROMPT_LENGTH).trim();	// name is whatever is after prompt
+			} else {
+				studentName = "missing";
+			}
+			//System.out.println("name: " + studentName);
+			line = br.readLine(); 											// read third line
+		} catch (FileNotFoundException e) {
+			System.err.println("Cannot find file " + submissionFileName);
+		} catch (IOException ioe) {
+			System.err.println("Cannot read from file " + submissionFileName);
+		}
+		return line;
+	}
+
+	// skip any blank lines and additional instructions for assignment (before first question instructor comments)
+	private String reachFirstQuestion(BufferedReader br, String line) throws SQLFEParseException {
+		line = Utilities.skipBlankLines(br, line);
+		//System.out.println("after any blanks, next line is: >" + line + "<");
+		
+		// if not already at the first question, look for any other instructor comments and trailing blanks and skip them
+		int attemptCount = 0;							// number of line attempts so far
+														// stop if user comments or start of new question
+        while (!Utilities.isUserCommentSingleLine(line) && !Utilities.isUserCommentMultiLineStart(line) && !Utilities.isQuestionFound(line) && 
+        		attemptCount <= MAX_LINE_ATTEMPTS) {						
+        	line = Utilities.skipInstructorComments(br, line);
```

Comment:
Will this not also skip lines that have a question number on them? This line seems like it will make the method skip through the entire submission.

Reply:
In the while check condition above this line, we are checking whether the line has question number (!Utilities.isQuestionFound(line)). So if it was a question it would never go inside this loop in the first place.